### ISSUE TITLE
XmlMapper overriding ObjectMapper in Spring Boot app

### ISSUE THREAD

**Issue by imrawa** [FasterXML/jackson-dataformat-xml#147]
Hi,

When I include a custom dependency **A** in my Spring Boot app **B** it causes restTemplate requests in **B** to serialize as XML instead of JSON. I think that this is due to a dependancy of **A** having the following in it's pom.xml dependancies:
```xml
<dependency>
	<groupId>com.fasterxml.jackson.dataformat</groupId>
	<artifactId>jackson-dataformat-xml</artifactId>
</dependency>
<dependency>
	<groupId>org.codehaus.woodstox</groupId>
	<artifactId>woodstox-core-asl</artifactId>
</dependency>
```

I'm guessing that these are causing Spring Boot to default to using XmlMapper instead of ObjectMapper. Is there anyway of preventing this? Ideally I would like to be able to contain XmlMapper to be used only withing **A** and not have to add any additional configuration on **B** (as the default is ObjectMapper)

Thanks!

**Comment by cowtowncoder** [comment]
This sounds more like a question to user or dev mailing lists; and most likely Spring Boot lists since it sounds like it is having trouble figuring out what data format to use.

**Comment by imrawa** [comment]
It would appear that this is due to the auto-configuring in Spring Boot. For some reason it is defaulting to using a mappingJackson2XmlHttpMessageConverter instead of mappingJackson2HttpMessageConverter when the request's Content-Type is not set (i.e. when using restTemplate.postForObject). I've raised this on Spring Boot Gitter, and will be raising an issue later.

Closing this as I do not believe that Spring Boot is responsible not Jackson
